typescript/schedules.md

```diff
@@ -0,0 +1,1246 @@
+   */
+  interceptors?: ScheduleClientInterceptors;
+
+  /**
+   * Identity to report to the server
+   *
+   * @default `${process.pid}@${os.hostname()}`
+   */
+  identity?: string;
+
+  /**
+   * Connection to use to communicate with the server.
+   *
+   * By default `ScheduleClient` connects to localhost.
+   *
+   * Connections are expensive to construct and should be reused.
+   */
+  connection?: ConnectionLike;
+
+  /**
+   * Server namespace
+   *
+   * @default default
+   */
+  namespace?: string;
+}
+
+export type ScheduleClientOptionsWithDefaults = Replace<
+  Required<ScheduleClientOptions>,
+  {
+    connection?: ConnectionLike;
+  }
+>;
+export type LoadedScheduleClientOptions = ScheduleClientOptionsWithDefaults & {
+  loadedDataConverter: LoadedDataConverter;
+};
+
+export function defaultScheduleClientOptions(): ScheduleClientOptionsWithDefaults {
+  return {
+    dataConverter: {},
+    // The equivalent in Java is ManagementFactory.getRuntimeMXBean().getName()
+    identity: `${process.pid}@${os.hostname()}`,
+    interceptors: {},
+    namespace: 'default',
+  };
+}
+
+// TODO
+// function ensureArgs<W extends Workflow, T extends ScheduleOptions<W>>(opts: T): Omit<T, 'args'> & { args: unknown[] } {
+//   const { args, ...rest } = opts;
+//   return { args: args ?? [], ...rest };
+// }
+
+interface SchdeduleHandleOptions extends GetSchdeduleHandleOptions {
+  workflowId: string;
+  runId?: string;
+  interceptors: ScheduleClientCallsInterceptor[];
+  /**
+   * A runId to use for getting the workflow's result.
+   *
+   * - When creating a handle using `getHandle`, uses the provided runId or firstExecutionRunId
+   * - When creating a handle using `start`, uses the returned runId (first in the chain)
+   * - When creating a handle using `signalWithStart`, uses the the returned runId
+   */
+  runIdForResult?: string;
+}
+
+/**
+ * Policy for overlapping Actions.
+ */
+export enum ScheduleOverlapPolicy {
+  /**
+   * Don't start a new Action.
+   */
+  Skip = 1,
+
+  /**
+   * Start another Action as soon as the current Action completes, but only buffer one Action in this way. If another
+   * Action is supposed to start, but one Action is running and one is already buffered, then only the buffered one will
+   * be started after the running Action finishes.
+   */
+  BufferOne,
+
+  /**
+   * Allows an unlimited number of Actions to buffer. They are started sequentially.
+   */
+  BufferAll,
+
+  /**
+   * Cancels the running Action, and then starts the new Action once the cancelled one completes.
+   */
+  CancelOther,
+
+  /**
+   * Terminate the running Action and start the new Action immediately.
+   */
+  TerminateOther,
+
+  /**
+   * Allow any number of Actions to start immediately.
+   *
+   * This is the only policy under which multiple Actions can run concurrently.
+   *
+   * {@link lastCompletionResult} and {@link lastFailure} will not be available.
```

Comment:
oh, does the proto still say that? I changed it so that they are (although the semantics if there are overlapping runs are slightly confusing.. there's an example in the docs)

Reply:
removing thanks